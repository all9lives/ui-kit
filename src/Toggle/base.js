import React, { PureComponent } from 'react'
import { css } from 'styled-components'
import Text from '../Text'
import View from '../View'
import { Checkbox } from '../Checkbox'

const base = css`
  position: relative;
`

export class Component extends PureComponent {
  handleValueChange = (value) => {
    if (this.props.onValueChange) {
      this.props.onValueChange(value)
    }
  }
  handleLabelPress = () => {
    if (this.props.onValueChange) {
      this.props.onValueChange(!this.props.value)
    }
  }
  render () {
    const { value, toggleStyle, label, labelStyle, tight, reverse } = this.props
    let [head, tail] = [
      <Text style={labelStyle} onPress={this.handleLabelPress}>{label}</Text>,
      <Checkbox style={toggleStyle} value={value} onValueChange={this.handleValueChange} />
    ]
    if (reverse) {
      [head, tail] = [tail, head]
    }
    return (
      <View align='middle' grow={1} direction='row'>
        {head}
        {!tight && <View grow={1} />}
        {tail}
      </View>
    )
  }
}

export default base
