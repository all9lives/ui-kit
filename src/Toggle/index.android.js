import styled from 'styled-components/native'
import base, { Component } from './base'
import defaultProps from './defaultProps'

const Toggle = styled(Component)`
  ${base}
`
Toggle.defaultProps = defaultProps

export default Toggle
