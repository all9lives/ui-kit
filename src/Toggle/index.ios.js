import Toggle from './index.android'

export default Toggle
