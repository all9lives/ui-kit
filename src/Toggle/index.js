import styled from 'styled-components'
import base, { Component } from './base'
import defaultProps from './defaultProps'

const Toggle = styled(Component)`
  ${base}
`
Toggle.defaultProps = defaultProps

export default Toggle
