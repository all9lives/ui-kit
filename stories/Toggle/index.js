import React from 'react'
import createStory from '../../storybook/createStory'
import Toggle from '../../src/Toggle'

createStory('Toggle', module)
  .add('default', () => <Toggle />)
  .add('with tight', () => <Toggle tight />)
  .add('with reverse', () => <Toggle reverse />)
  .add('with label:Toggle', () => <Toggle label='Toggle' />)
  .add('with labelStyle', () => <Toggle labelStyle={{ color: 'blue' }} />)
  .add('with toggleStyle', () => <Toggle toggleStyle={{ backgroundColor: 'blue' }} />)
