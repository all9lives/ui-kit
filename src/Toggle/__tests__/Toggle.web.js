/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Toggle from '../index.js'

describe('Toggle.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Toggle />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
