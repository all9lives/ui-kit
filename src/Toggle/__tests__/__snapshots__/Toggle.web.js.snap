// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle.web renders a snapshot 1`] = `
.c0 {
  position: relative;
}

<Component
  className="c0"
  color="#191919"
  label="Label"
  size={16}
/>
`;
